src/v/debug_bundle/debug_bundle_service.cc

```diff
@@ -905,4 +948,71 @@ ss::future<> service::maybe_reload_previous_run() {
     }
 }
 
+ss::future<> service::tick() {
+    auto units = co_await _process_control_mutex.get_units();
+    if (
+      !_debug_bundle_cleanup_binding().has_value() || _rpk_process == nullptr
+      || _rpk_process->process_status() != debug_bundle_status::success) {
+        co_return;
+    }
+
+    auto cleanup_seconds = _debug_bundle_cleanup_binding().value();
+    auto cleanup_timepoint = _rpk_process->finished_time()
+                             + std::chrono::seconds{cleanup_seconds};
+
+    if (cleanup_timepoint <= clock::now()) {
+        vlog(
+          lg.info,
+          "Cleaning up debug bundle for job {}",
+          _rpk_process->job_id());
+        try {
+            co_await cleanup_previous_run();
+        } catch (const std::exception& e) {
+            vlog(
+              lg.warn,
+              "Failed to clean up previous run for job {}: {}",
+              _rpk_process->job_id(),
+              e.what());
+        }
+        _rpk_process->set_expired();
+    }
```

Comment:
maybe i'm missing something, but it seems fundamentally this could be simplified to:

```
tick():
  - if not file_exists(debug_bundle): return
  - if ctime(debug_bundle) > max_age: delete_file(debug_bundle)

on_boot():
  - tick()
  - arm_periodic(max_age, tick)
```

one thing that caught my eye was

```
    auto cleanup_timepoint = _rpk_process->finished_time()
                             + std::chrono::seconds{cleanup_seconds};
```

which i found surprising, like a debug bundle exists on disk and has an age, why would i be interested in when some instance of rpk debug bundle completed?

Reply:
> like a debug bundle exists on disk and has an age

Can we/do we trust the timestamp of the file on disc?  Do any other processes in RP trust/use the reported age of files on disk?  Is it possible for that timestamp to change between reboots of a k8s node?  I ask because I don't know.  I save off the timestamp of when the process finished which should fairly represent the age of the debug bundle.

Cleanup also involves cleaning up the file, the process output, and the kvstore.

Reply:
>I save off the timestamp of when the process finished which should fairly represent the age of the debug bundle.

how does the expiration work across reboots?

Reply:
when I reload metadata I look at the stored end time, if the timeout elapsed while the system was down, I clean up the files and remove the metadata from the kvstore